src/main/java/mate/academy/springboot/datajpa/service/CategoryServiceImpl.java

```diff
@@ -0,0 +1,35 @@
+package mate.academy.springboot.datajpa.service;
+
+import java.util.NoSuchElementException;
+import mate.academy.springboot.datajpa.model.Category;
+import mate.academy.springboot.datajpa.repository.CategoryRepository;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository categoryRepository;
+
+    public CategoryServiceImpl(CategoryRepository categoryRepository) {
+        this.categoryRepository = categoryRepository;
+    }
+
+    @Override
+    public Category save(Category category) {
+        return categoryRepository.save(category);
+    }
+
+    @Override
+    public Category getById(Long id) {
+        return categoryRepository.findById(id).orElseThrow(NoSuchElementException::new);
+    }
+
+    @Override
+    public void deleteById(Long id) {
+        categoryRepository.deleteById(id);
+    }
+
+    @Override
+    public Category update(Category category) {
+        return categoryRepository.save(category);
+    }
```

Comment:
here you have same logic as in save method, i think we can use save() method for both purposes
![image](https://user-images.githubusercontent.com/111656381/210748412-343e645a-0790-4aaf-9b69-9c177c8cc8a0.png)
same in ProductServiceImpl